src/testRunner/unittests/tsbuildWatch/publicApiImpliedNodeFormat.ts

```diff
@@ -0,0 +1,254 @@
+export class c { }
+export enum e { }
+// leading
+export function f2() { } // trailing`
+        };
+        const webpackConfig: File = {
+            path: `${projectRoot}/webpack/tsconfig.json`,
+            content: JSON.stringify({
+                compilerOptions: { composite: true, },
+                references: [{ path: "../shared/tsconfig.json" }]
+            })
+        };
+        const webpackIndex: File = {
+            path: `${projectRoot}/webpack/index.ts`,
+            content: `export function f2() { }
+export class c2 { }
+export enum e2 { }
+// leading
+export function f22() { } // trailing`
+        };
+        const commandLineArgs = ["--b", "--w", /*"--verbose"*/];
+        const { sys, baseline, oldSnap, cb, getPrograms } = createBaseline(createWatchedSystem([libFile, { ...libFile, path: "/a/lib/lib.es2020.d.ts" }, solution, sharedConfig, sharedConfigOptions, sharedPackageJson, sharedIndex, webpackConfig, webpackIndex], { currentDirectory: projectRoot }));
+
+        const writeToConsole = false; // MUST BE false WHEN NOT TESTING SINGULARLY
+        if (writeToConsole){
+            const origSysWrite = sys.write.bind(sys);
+            sys.write = (message: string)=>{
+                console.log(message);
+                origSysWrite(message);
+            };
+        }
+        type ImpliedNodeFormat = ModuleKind.CommonJS | ModuleKind.ESNext | undefined;
+        const impliedNodeFormatToString = (f: ImpliedNodeFormat)=>(f===undefined)? "undefined" : ModuleKind[f];
+        const lastImpliedNodeFormats = new Map<string,{impliedNodeFormat: ImpliedNodeFormat /*, gotImpledNodeFormat: ImpliedNodeFormat*/}>();
+        function printLastImpliedNodeFormats(){
+            lastImpliedNodeFormats.forEach((x,fileName)=>{
+                sys.write(`fileName:${fileName},impliedNodeFormat:${impliedNodeFormatToString(x.impliedNodeFormat)}`+sys.newLine);
+            });
+        }
+        // function printModuleResolutionCache(buildr: SolutionBuilder<EmitAndSemanticDiagnosticsBuilderProgram>){
+        //     const x = { moduleResolutionCache: buildr.getModuleResolutionCache() };
+        //     sys.write(JSON.stringify(x,undefined,2));
+        // }
+        const buildHost = createSolutionBuilderWithWatchHostForBaseline(sys, cb);
+        buildHost.getCustomTransformers = getCustomTransformers;
+        const builder = createSolutionBuilderWithWatch(buildHost, [solution.path], { verbose: true });
+        const compilerOptions: {
+            lib: ["es2020"],
+            target: "es2020"
+            moduleResolution?: "Node12" | "NodeNext", // absolutely needed
+            module?: "Node12" | "NodeNext"
+        }={
+            lib: ["es2020"],
+            moduleResolution: "Node12",
+            target: "es2020",
+        };
+
+        builder.build();
+        runWatchBaseline({
+            scenario: "publicApi",
+            subScenario: "with custom transformers, test that impliedNodeFormat is set",

```

Comment:
I don't know how good filesystems are at handling `,` -- I feel like the escaping code in `runWatchBaseline` should remove it.

Reply:
Yes it comes out ugly `with-custom-transformers,-test-that-impliedNodeFormat-is-set`.  I'll normalize it.,

Reply:
Change to 
```
subScenario: "with custom transformers impliedNodeFormat",
```